onboarding/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	runtime "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	onboardingTicketPublicKeySecretName  = "onboarding-ticket-key" //Name of existing public key which is used ocs-operator
+	onboardingTicketPrivateKeySecretName = "onboarding-ticket-private-key"
+	serviceAccountName                   = "onboarding-secret-generator"
+)
+
+func main() {
+
+	clientset, err := newClient()
+	if err != nil {
+		klog.Error(err, "failed to create controller-runtime client")
+		return
+	}
+
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		klog.Error(err, "unable to get operator namespace")
+		os.Exit(1)
+	}
+
+	// 1. Check public key secret exist or not
+	_, err = clientset.CoreV1().Secrets(operatorNamespace).Get(context.TODO(), onboardingTicketPublicKeySecretName, metav1.GetOptions{})
+
+	if err != nil && kerrors.IsNotFound(err) {
+
+		// Generate RSA key.
+		var err error
+		privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+		if err != nil {
+			panic(err)
+		}
+
+		publicKey := &privateKey.PublicKey
+
+		// Export the keys to pem string
+		privatePem := convertRsaPrivateKeyAsPemStr(privateKey)
+		publicPem, err := convertRsaPublicKeyAsPemStr(publicKey)
+
+		if err != nil {
+			panic(err)
+		}
+
+		privateSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        onboardingTicketPrivateKeySecretName,
+				Namespace:   operatorNamespace,
+				Annotations: map[string]string{"kubernetes.io/service-account.name": serviceAccountName},
+			},
+			Type: "kubernetes.io/service-account-token",
+			StringData: map[string]string{
+				"key": privatePem,
+			},
+		}
+
+		_, err = clientset.CoreV1().Secrets(operatorNamespace).Create(context.Background(), privateSecret, metav1.CreateOptions{})
```

Comment:
- this needs a bit of discussion w/ Ohad about whether to use a finalizer for this private secret to protect against deletion or not

Reply:
I will check this ohad and based on that i will add finalizer.

Reply:
I just checked w/ him, no need. rest api will delete the secret when we want to rotate it